File: reuse_java/src/test/java/com/kmmania/JumperTest.java
package com.kmmania;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

class JumperTest {
    @Test
    void display_DisplaysCorrectInformation() {
        // Given
        String firstName = "Alice";
        String lastName = "Smith";
        String speciality = "Distance";
        AthleteInterface athleteMock = mock(AthleteInterface.class);
        when(athleteMock.mySpeciality()).thenReturn("My speciality is " + speciality + ".");
        Jumper jumper = new Jumper(firstName, lastName, speciality);
        jumper.setAthlete(athleteMock);

        // When
        jumper.display();

        // Then
        assertThat(jumper.getFirstName()).isEqualTo(firstName);
        assertThat(jumper.getLastName()).isEqualTo(lastName);
        // verify(athleteMock).mySpeciality();
    }
}

File: reuse_kotlin/src/test/kotlin/JumperTest.kt
import com.kmmania.AthleteInterface
import com.kmmania.Jumper
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.mockito.Mockito.*

class JumperTest {

    @Test
    fun `display() method should display correct information`() {
        // Given
        val firstName = "Marco"
        val lastName = "Tamberi"
        val speciality = "high jump"
        val athleteMock = mock(AthleteInterface::class.java)
        `when`(athleteMock.mySpeciality()).thenReturn("My speciality is $speciality.")
        val jumper = Jumper(firstName, lastName, athleteMock)

        // When
        jumper.display()

        // Then
        assertThat(jumper.firstName).isEqualTo(firstName)
        assertThat(jumper.lastName).isEqualTo(lastName)
        verify(athleteMock).mySpeciality()
    }
}